app/models/concerns/client_image_consumer.rb

```diff
@@ -10,11 +10,15 @@ module ClientImageConsumer
     has_many :client_files unless respond_to?(:client_files)
     has_many :source_eto_client_lookups, through: :source_clients, source: :eto_client_lookups
 
-    # Returns actual image bytes. Cache time limit not yet implemented.
+    # Cache time limit not yet implemented
+    # @return [String, nil] actual image bytes.
     def image(_cache_for = 10.minutes)
+      # If we call `image` on a source client, return the attached image
+      return image_for_source_client if source?
+
       # Check first for locally uploaded image or a cached ETO image
-      # Otherwise, check for an image connected to the source client
-      local_client_image_data || source_clients.detect(&:image_for_source_client)&.image_for_source_client
+      # Otherwise, check for an image connected to any source client
+      local_client_image_data || source_clients.detect { |sc| sc.image_for_source_client.length > 100 }&.image_for_source_client
```

Comment:
looks like image_for_source_client can return nil, maybe protect against that? otherwise calling length will crash
```suggestion
      local_client_image_data || source_clients.detect { |sc| (sc.image_for_source_client || '').length > 100 }&.image_for_source_client
```